### ISSUE TITLE
Update spec draft according to change made (if approved by majority) #GraphQL

### ISSUE THREAD

**Issue by thejibz** [eclipse/microprofile-sandbox#32]
Changes:
- Put a "value" field in every annotations.
- Merging Type and FieldsOrder annotations.
- Merging InputType and InputFieldsOrder annotations.
- Adding deprecationReason in:
		@Type
		@InputType
		@Field
		@InputField

**Comment by andymc12** [comment]
> Put a "value" field in every annotations.

We decided this was appropriate, where applicable. "value" is preferred to "name" to avoid extra code.

> Merging Type and FieldsOrder annotations.
> Merging InputType and InputFieldsOrder annotations.

Last week we decided to defer this decision - between `@FieldsOrder`/`@InputFieldsOrder` on the type vs `Order(1)` on the fields.  This could be decided in the v2 timeframe.

> Adding deprecationReason in
> @type
> @InputType
> @field
> @InputField

Last week, we decided this should be a separate `@Deprecated("reason")` annotation - where the reason should be defaulted to a value like "Deprecated" to avoid issues with some tooling.
